tensorboard/plugins/projector/polymer3/vz_projector/vz-projector-data-panel.ts

```diff
@@ -12,810 +12,746 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-      superviseColumn: {type: String},
-      superviseColumnChanged: {type: Object},
-      showSuperviseSettings: {type: Boolean, value: false},
-    },
-    observers: ['_generateUiForNewCheckpointForRun(selectedRun)'],
-  });
-
-  export class DataPanel extends DataPanelPolymer {
-    selectedLabelOption: string;
-    selectedColorOptionName: string;
-    showForceCategoricalColorsCheckbox: boolean;
-    showSuperviseSettings: boolean;
-
-    private normalizeData: boolean;
-    private labelOptions: string[];
-    private colorOptions: ColorOption[];
-    forceCategoricalColoring: boolean = false;
-
-    private metadataEditorInput: string;
-    private metadataEditorInputLabel: string;
-    private metadataEditorButtonDisabled: boolean;
-    private superviseInput: string;
-    private superviseInputLabel: string;
-    private superviseInputSelected: string;
-    private superviseColumn: string;
-
-    private selectedPointIndices: number[];
-    private neighborsOfFirstPoint: knn.NearestEntry[];
-    private selectedTensor: string;
-    private selectedRun: string;
-    private dataProvider: DataProvider;
-    private tensorNames: {name: string; shape: number[]}[];
-    private runNames: string[];
-    private projector: Projector;
-    private projectorConfig: ProjectorConfig;
-    private colorLegendRenderInfo: ColorLegendRenderInfo;
-    private spriteAndMetadata: SpriteAndMetadataInfo;
-    private metadataFile: string;
-
-    ready() {
-      super.ready();
-      this.normalizeData = true;
-      this.superviseInputSelected = '';
-    }
-
-    initialize(projector: Projector, dp: DataProvider) {
-      this.projector = projector;
-      this.dataProvider = dp;
-      this.setupUploadButtons();
-
-      // Tell the projector whenever the data normalization changes.
-      // Unknown why, but the polymer checkbox button stops working as soon as
-      // you do d3.select() on it.
-      this.$$('#normalize-data-checkbox').addEventListener('change', () => {
-        this.projector.setNormalizeData(this.normalizeData);
-      });
-
-      let forceCategoricalColoringCheckbox = this.$$(
-        '#force-categorical-checkbox'
+import {PolymerElement} from '@polymer/polymer';
+import {LegacyElementMixin} from '@polymer/polymer/lib/legacy/legacy-element-mixin';
+import {customElement, property, observe} from '@polymer/decorators';
+
+import * as d3 from 'd3';
+
+import '@polymer/paper-button';
+import '@polymer/paper-checkbox';
+import '@polymer/paper-dialog-scrollable';
+import '@polymer/paper-dialog';
+import '@polymer/paper-dropdown-menu/paper-dropdown-menu';
+import '@polymer/paper-icon-button';
+import '@polymer/paper-input/paper-input';
+import '@polymer/paper-item';
+import '@polymer/paper-listbox';
+import '@polymer/paper-tooltip';
+
+import {Projector} from './vz-projector';
+import {template} from './vz-projector-data-panel.html';
+import {
+  ColorLegendThreshold,
+  ColorLegendRenderInfo,
+} from './vz-projector-legend';
```

Comment:
just checking, since ColorLegendThreshold/RenderInfo are TS types, do we need an additional "import './vz-projector-legend'" to get the custom element?

Reply:
Added one.